src/bijectors/scale.jl

```diff
@@ -7,27 +7,21 @@ function Scale(a::Union{Real,AbstractArray}; dim::Val{D} = Val(ndims(a))) where
 end
 
 (b::Scale)(x) = b.a .* x
-(b::Scale{<:Real})(x::AbstractArray) = b.a .* x
 (b::Scale{<:AbstractMatrix})(x::AbstractVecOrMat) = b.a * x
-(b::Scale{<:AbstractVector{<:Real}, 2})(x::AbstractMatrix{<:Real}) = b.a .* x
-
-(ib::Inversed{<:Scale})(y) = Scale(inv(ib.orig.a))(y)
-(ib::Inversed{<:Scale{<:AbstractVector}})(y) = Scale(inv.(ib.orig.a))(y)
-function (ib::Inversed{<:Scale{<:AbstractMatrix, 1}})(y::AbstractVecOrMat)
+(ib::Inverse{<:Scale})(y) = Scale(inv(ib.orig.a))(y)
+(ib::Inverse{<:Scale{<:AbstractVector}})(y) = Scale(inv.(ib.orig.a))(y)
+function (ib::Inverse{<:Scale{<:AbstractMatrix, 1}})(y::AbstractVecOrMat)
     return ib.orig.a \ y
 end
 
-# inv(b::Scale{T, D}) where {T, D} = Scale(inv(b.a); dim = Val(D))
-# inv(b::Scale{<:AbstractVector, D}) where {D} = Scale(inv.(b.a); dim = Val(D))
-
 # We're going to implement custom adjoint for this
 logabsdetjac(b::Scale{T, N}, x) where {T, N} = _logabsdetjac_scale(b.a, x, Val(N))
 
 _logabsdetjac_scale(a::Real, x::Real, ::Val{0}) = log(abs(a))
 _logabsdetjac_scale(a::Real, x::AbstractVector, ::Val{0}) = fill(log(abs(a)), length(x))
 _logabsdetjac_scale(a::Real, x::AbstractVector, ::Val{1}) = log(abs(a)) * length(x)
 _logabsdetjac_scale(a::Real, x::AbstractMatrix, ::Val{1}) = fill(log(abs(a)) * size(x, 1), size(x, 2))
-_logabsdetjac_scale(a::AbstractVector, x::AbstractVector, ::Val{1}) = sum(log.(abs.(a)))
-_logabsdetjac_scale(a::AbstractVector, x::AbstractMatrix, ::Val{1}) = fill(sum(log.(abs.(a))), size(x, 2))
-_logabsdetjac_scale(a::AbstractMatrix, x::AbstractVector, ::Val{1}) = log(abs(det(a)))
-_logabsdetjac_scale(a::AbstractMatrix, x::AbstractMatrix{T}, ::Val{1}) where {T} = log(abs(det(a))) * ones(T, size(x, 2))
\ No newline at end of file
+_logabsdetjac_scale(a::AbstractVector, x::AbstractVector, ::Val{1}) = sum(x -> log(abs(x)), a)
+_logabsdetjac_scale(a::AbstractVector, x::AbstractMatrix, ::Val{1}) = fill(sum(x -> log(abs(x)), a), size(x, 2))
+_logabsdetjac_scale(a::AbstractMatrix, x::AbstractVector, ::Val{1}) = logabsdet(a)
```

Comment:
`logabsdet` returns a 2-tuple, right? So should this be `logabsdet(a)[1]`?

Reply:
Yes I think so. Nice catch!